backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/DiagramService.java

```diff
@@ -48,29 +53,52 @@
 @Service
 public class DiagramService implements IDiagramService {
 
+    private final IObjectService objectService;
+
     private final IRepresentationRepository representationRepository;
 
     private final ObjectMapper objectMapper;
 
     private final Logger logger = LoggerFactory.getLogger(DiagramService.class);
 
-    public DiagramService(IRepresentationRepository representationRepository, ObjectMapper objectMapper) {
+    public DiagramService(IObjectService objectService, IRepresentationRepository representationRepository, ObjectMapper objectMapper) {
+        this.objectService = Objects.requireNonNull(objectService);
         this.representationRepository = Objects.requireNonNull(representationRepository);
         this.objectMapper = Objects.requireNonNull(objectMapper);
     }
 
     @Override
     public Diagram create(DiagramCreationParameters parameters) {
+        return this.create(parameters, Optional.empty());
+    }
+
+    @Override
+    public Diagram create(DiagramCreationParameters parameters, Diagram previousDiagram) {
+        return this.create(parameters, Optional.of(previousDiagram));
+    }
+
+    private Diagram create(DiagramCreationParameters parameters, Optional<Diagram> optionalPreviousDiagram) {
+        DiagramDescription diagramDescription = parameters.getDiagramDescription();
+
         VariableManager variableManager = new VariableManager();
         variableManager.put(GetOrCreateRandomIdProvider.PREVIOUS_REPRESENTATION_ID, parameters.getId());
         variableManager.put(DiagramDescription.LABEL, parameters.getLabel());
         variableManager.put(VariableManager.SELF, parameters.getObject());
         variableManager.put(IEditingContext.EDITING_CONTEXT, parameters.getEditingContext());
 
-        DiagramComponentProps props = new DiagramComponentProps(variableManager, parameters.getDiagramDescription());
+        DiagramComponentProps props;
+        if (optionalPreviousDiagram.isPresent()) {
+            Diagram previousDiagram = optionalPreviousDiagram.get();
+            if (this.containsUnsyncDescriptions(diagramDescription)) {
+                Map<String, Object> targetSemanticObjectsMap = this.getSemanticElements(previousDiagram, parameters.getEditingContext());
+                variableManager.put(DiagramDescription.TARGET_SEMANTIC_OBJECTS, targetSemanticObjectsMap);
+            }
+            props = new DiagramComponentProps(variableManager, diagramDescription, previousDiagram);
```

Comment:
There are two ways to give information to a component, using the props directly or thanks to the variable manager. Both ways have specific meanings and the variable manager should not be considered as "the way to give something recursively to every components". It's only data which will be used by the various providers of the descriptions such as the semantic candidates provider or the target object id provider.

In other words, adding something to the variable manager means adding a new public API that specifiers can rely on to create their providers and expressions if they rely on the compatibility layer. The `targetSemanticObjectsMap` should not be visible to specifiers.